File: libsql/src/main/java/tech/turso/libsql/EmbeddedReplicaDatabase.java
package tech.turso.libsql;

public class EmbeddedReplicaDatabase extends Database {
    EmbeddedReplicaDatabase(long ptr) throws Exception {
        super(ptr);
    }

    public void sync() {
        nativeSync(nativePtr);
    }

    private static native void nativeSync(long ptr);
}

File: libsql/src/main/kotlin/tech/turso/libsql/EmbeddedReplicaDatabase.kt
package tech.turso.libsql

class EmbeddedReplicaDatabase(inner: Long) : Database(inner) {
    fun sync() {
        require(this.inner != 0L) { "Attempted to sync a closed Database" }
        nativeSync(this.inner)
    }

    private external fun nativeSync(db: Long)
}
